File: bbms_frontend_v2/bbms_v2/src/services/superAdminService.js
// src/services/superAdminService.js
import axios from 'axios';

const API_URL = import.meta.env.VITE_API_URL || 'http://localhost:8080/api';

// ✅ Get all admins
export const getAllAdmins = async () => {
    try {
        const response = await axios.get(`${API_URL}/admins`);
        return response.data; // List<UserDTO with role=ADMIN
    } catch (error) {
        throw error.response?.data || 'Failed to fetch admins';
    }
};

// ✅ Create new admin
export const createNewAdmin = async (adminData) => {
    try {
        const response = await axios.post(`${API_URL}/admins`, { ...adminData, role: 'ADMIN' });
        return response.data; // UserDTO
    } catch (error) {
        throw error.response?.data || 'Failed to create admin';
    }
};

// ✅ Delete admin
export const deleteAdminById = async (id) => {
    try {
        await axios.delete(`${API_URL}/admins/${id}`);
        return true;
    } catch (error) {
        throw error.response?.data || 'Failed to delete admin';
    }
};

// ✅ System stats
export const getSystemStats = async () => {
    try {
        const response = await axios.get(`${API_URL}/system/stats`);
        return response.data; // { users, staff, buses, routes }
    } catch (error) {
        throw error.response?.data || 'Failed to fetch system stats';
    }
};

File: bbms_backend_v2/baust_trip_connect_v2/src/main/java/com/baust/baust_trip_connect_v2/service/SuperAdminService.java
package com.baust.baust_trip_connect_v2.service;

import com.baust.baust_trip_connect_v2.dto.UserDTO;
import com.baust.baust_trip_connect_v2.model.Role;
import com.baust.baust_trip_connect_v2.model.User;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Map;

@Service
public class SuperAdminService {

    @Autowired
    private UserService userService;
    @Autowired
    private BusService busService;
    @Autowired
    private RouteService routeService;

    // ✅ Get all admins
    public List<UserDTO> getAllAdmins() {
        return userService.getAllUsers().stream()
                .filter(u -> u.getRole() == Role.ADMIN)
                .toList();
    }

    // ✅ Create new admin
    public UserDTO createAdmin(User user) {
        user.setRole(Role.ADMIN);
        User saved = userService.saveUser(user);
        return userService.toDTO(saved);
    }

    // ✅ Delete admin
    public void deleteAdmin(Long id) {
        userService.deleteUser(id);
    }

    // ✅ System stats
    public Map<String, Long> getSystemStats() {
        long users = userService.getAllUsers().size();
        long staff = userService.getAllUsers().stream().filter(u -> u.getRole() == Role.STAFF).count();
        long admins = userService.getAllUsers().stream().filter(u -> u.getRole() == Role.ADMIN).count();
        long buses = busService.getAllBuses().size();
        long routes = routeService.getAllRoutes().size();

        return Map.of(
                "users", users,
                "staff", staff,
                "admins", admins,
                "buses", buses,
                "routes", routes
        );
    }
}
